src/remote/client/element-finder-remote.ts

```diff
@@ -0,0 +1,212 @@
+    const response = await this.executeEndpoint('POST', 'element/query/value', {locators: this.locators});
+    const body = await response.json();
+    return body.value;
+  }
+
+  public async getHTML(includeSelectorTag: boolean): Promise<string> {
+    const response = await this.executeEndpoint('POST', 'element/query/html', {
+      locators: this.locators,
+      includeSelectorTag
+    });
+    const body = await response.json();
+    return body.html;
+  }
+
+  public async getAttribute(name: string): Promise<string> {
+    const response = await this.executeEndpoint('POST', 'element/query/attribute', {locators: this.locators, name});
+    const body = await response.json();
+    return body.attribute;
+  }
+
+  public async getCSSProperty(name: string): Promise<string> {
+    const response = await this.executeEndpoint('POST', 'element/query/css-property', {locators: this.locators, name});
+    const body = await response.json();
+    return body.property;
+  }
+
+  public async getProperty(name: string): Promise<any> {
+    const response = await this.executeEndpoint('POST', 'element/query/property', {locators: this.locators, name});
+    const body = await response.json();
+    return body.property;
+  }
+
+  public async getBoundingRect(): Promise<{ x: number, y: number, width: number, height: number }> {
+    const response = await this.executeEndpoint('POST', 'element/query/bounding-rect', {locators: this.locators});
+    const body = await response.json();
+    return body.boundingRect;
+  }
+
+  public async getPosition(): Promise<{ x: number, y: number }> {
+    const response = await this.executeEndpoint('POST', 'element/query/position', {locators: this.locators});
+    const body = await response.json();
+    return body.position;
+  }
+
+  public async getSize(): Promise<{ width: number, height: number }> {
+    const response = await this.executeEndpoint('POST', 'element/query/size', {locators: this.locators});
+    const body = await response.json();
+    return body.size;
+  }
+
+
+  // Actions
+  public async click(): Promise<void> {
+    await this.executeEndpoint('POST', 'element/action/click', {locators: this.locators});
+  }
+
+  public async moveTo(): Promise<void> {
+    await this.executeEndpoint('POST', 'element/action/move-to', {locators: this.locators});
+  }
+
+  public async clear(): Promise<void> {
+    await this.executeEndpoint('POST', 'element/action/move-to', {locators: this.locators});
+  }
+
+  public async write(text: string): Promise<void> {
+    await this.executeEndpoint('POST', 'element/action/write', {locators: this.locators, text});
+  }
+
+  public async tap(): Promise<void> {
+    await this.executeEndpoint('POST', 'element/action/tap', {locators: this.locators});
+  }
+
+
+  // Condition waits
+  public async waitToBePresent(timeout: number = 500): Promise<void> {
+    await this.executeEndpoint('GET', 'element/wait/present', {locators: this.locators});
```

Comment:
This wait GET call should be updated to be a POST call. As we need to pass a request body (i.e locators), the HTTP protocol does not support so for GET method. As we do in other cases, this should be changed to be a POST.
This applies to all condition waits.

Reply:
Fixed